cli/src/factory_ops/unboxing/network/public_mgmt

```diff
@@ -0,0 +1,253 @@
+        if [[ ${remote} == ${pvt_ip_a} ]]; then
+            srvnode=srvnode-1
+            node_name=Server-A
+        elif [[ ${remote} == ${pvt_ip_b} ]]; then
+            srvnode=srvnode-2
+            node_name=Server-B
+        else
+            # This should never be hit
+            _lerror "This should never be seen. If you see this, there is something unexpected heppening."
+            exit 23
+        fi
+
+        mgmt_if=$(get_pillar_data cluster:${srvnode}:network:mgmt_nw:iface:0)
+        _linfo "Setting DHCP on ${srvnode} for interface: ${mgmt_if}"
+        backup_ifcfg ${remote} ${mgmt_if}
+        
+        _linfo "Setting /etc/resolv.conf on ${node_name} for interface: ${mgmt_if}"
+        ssh_over_pvt_data ${remote} "salt-call --local state.apply components.system.network ${SALT_OPTS}"
+        _linfo "Setting ifcfg-${mgmt_if} on ${node_name}"
+        ssh_over_pvt_data ${remote} "salt-call --local state.apply components.system.network.mgmt.public ${SALT_OPTS}"
+
+    done
+}
+
+
+# 2.  If DHCP
+#     2.1.  Set cluster entries for public management IP to None
+#     2.2.  Update /etc/resolv.conf using dhclient
+function set_mgmt_dhcp_config {
+    _linfo "================================================================================"
+    _linfo "Setting Management Network to DHCP"
+    _linfo "================================================================================"
+
+    # Reset cluster value for management network
+    _linfo "    "
+    _linfo "You have selected to proceed with DHCP based configuration for management network interface."
+    _linfo "    "
+    proceed_check
+    _linfo "    "
+
+    # srvnode-1
+    _linfo "Preparing cluster pillar for setting management network configuration to DHCP."    
+    # Update pillar
+    update_cluster_pillar_for_mgmt_ips
+
+    # Set network config to DHCP for pillar value cluster:srvnode-1:network:mgmt_nw:iface:0
+    set_mgmt_nw_config
+    
+    # Sanity check
+    test_mgmt_ips
+
+    _linfo "================================================================================"
+    _linfo "        "
+}
+
+
+# 3.  If Static
+#     3.1.  IP address for public management interface (E.g. for eno1) <Mandatory>:
+#           for srvnode-1 & srvnode-2
+#     3.2.  Gateway for management network <Mandatory>
+#     3.3.  BMC IP <Mandatory>
+#           for srvnode-1 & srvnode-2
+function set_mgmt_static_config {
+    _linfo "================================================================================"
+    _linfo "Setting Management Network to Static"
+    _linfo "================================================================================"
+    _linfo "    "
+
+    _linfo "You have selected to proceed with static IP based configuration for management network interface."
+    _linfo "    "
+
+    mgmt_if_1=$(get_pillar_data cluster:srvnode-1:network:mgmt_nw:iface:0)
+    mgmt_if_2=$(get_pillar_data cluster:srvnode-1:network:mgmt_nw:iface:0)
```

Comment:
@ypise  shoudn't it be `cluster:srvnode-2:network:mgmt_nw:iface:1` here?

Reply:
Fixed